grid2op/Backend/pandaPowerBackend.py

```diff
@@ -1024,14 +1026,14 @@ def _aux_runpf_pp(self, is_dc: bool):
             # else:
             #     self._pf_init = "auto"
 
-            if (~self._grid.load["in_service"]).any():
+            if not self.allow_shedding and (~self._grid.load["in_service"]).any():
                 # TODO see if there is a better way here -> do not handle this here, but rather in Backend._next_grid_state
                 raise pp.powerflow.LoadflowNotConverged("Disconnected load: for now grid2op cannot handle properly"
                                                         " disconnected load. If you want to disconnect one, say it"
                                                         " consumes 0. instead. Please check loads: "
                                                         f"{(~self._grid.load['in_service'].values).nonzero()[0]}"
                                                         )
-            if (~self._grid.gen["in_service"]).any():
+            if not self.allow_shedding and (~self._grid.gen["in_service"]).any():
                 # TODO see if there is a better way here -> do not handle this here, but rather in Backend._next_grid_state
                 raise pp.powerflow.LoadflowNotConverged("Disconnected gen: for now grid2op cannot handle properly"
                                                         " disconnected generators. If you want to disconnect one, say it"
```

Comment:
This should be rather done in the function used by the Environment, that is the `next_grid_state` which in turn calls the `_runpf_with_diverging_exception` function.

So I think all these checks (same for check in DC by the way) should rather be done within the `_runpf_with_diverging_exception` function and not here :)

Reply:
(to be more explicit this should be in the `Backend`, madeby reading data from the backend (instead of using the `self._grid`) and removed from the `PandaPowerBackend`)

Reply:
Do you mean I should get rid of the original (i.e. not something I wrote) `if (~self._grid.gen["in_service"]).any():` check within the PandaPowerBackend as well then (i.e. move all these into the Backend class?)

Reply:
Yes exactly, all these checks have nothing to do in the specific backend implementation (in this case PandaPowerBackend) but should be moved to the base backend class (Backend).
And when doing this, these checks should be made "backend implementation" independant, so use only function of the public "backend" api.

Reply:
Had to think a bit how to check this in a backend-agnostic way, since there is no 'in_service' for others. My solution was to use the topo_vect and check if any bus was -1 (detached). Now the check is in the _runpf_with_diverging_exception method